cap-async-std/src/fs_utf8/mod.rs

```diff
@@ -39,18 +39,20 @@ fn from_utf8<P: AsRef<str>>(path: P) -> std::io::Result<std::path::PathBuf> {
     // in the future, the idea is we could avoid this.
     let string = arf_strings::PosixString::from_path_str(path.as_ref())
         .map_err(|_| std::io::Error::new(std::io::ErrorKind::InvalidData, "invalid path string"))?;
-    let bytes = string.as_cstr().to_bytes();
 
     #[cfg(any(unix, target_os = "redox", target_os = "wasi"))]
     let path = {
         use std::{ffi::OsStr, os::unix::ffi::OsStrExt};
+        let bytes = string.as_cstr().to_bytes();
         OsStr::from_bytes(bytes).to_owned()
     };
 
     #[cfg(windows)]
     let path = {
-        use std::{ffi::OsString, os::windows::prelude::*};
-        OsString::from_wide(bytes)
+        use std::{ffi::OsString, os::windows::ffi::OsStringExt};
+        let utf8 = string.as_cstr().to_string_lossy(); // This is OK since we went through arf_strings first.
```

Comment:
This is coming out the "back end" of arf_strings, so `string` isn't guaranteed to be valid UTF-8; it's an arbitrary NUL-terminated byte sequence. I think this whole area is something we're going to have to figure out within the arf-strings library itself. Probably we'll need a `WindowsString` counterpart to `PosixString`.

For now, to be safe, instead of `to_string_lossy()`, how about `to_str()?`. That way we won't risk accidental collisions due to lossiness.

Reply:
Ah ok. I wasn’t sure about this one. I’ll be happy to add the missing bits to arf-strings to support utf16. In the meantime, I’ll tweak this to use lossless `to_str`.